pyvo/mivot/viewer/xml_viewer.py

```diff
@@ -0,0 +1,169 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+XMLViewer provides several getters on XML instances built by
+ `~pyvo.mivot.viewer.mivot_viewer`.
+"""
+from pyvo.mivot.utils.exceptions import MivotElementNotFound
+from pyvo.mivot.utils.xpath_utils import XPath
+from pyvo.utils.prototype import prototype_feature
+
+
+@prototype_feature('MIVOT')
+class XMLViewer:
+    """
+    The XMLViewer  is used by `~pyvo.mivot.viewer.mivot_viewer`
+    to extract from the XML serialization of the model,
+    elements that will be used to build the dictionary from which
+    the Python class holding the mapped model will be generated.
+    """
+    def __init__(self, xml_view):
+        self._xml_view = xml_view
+
+    @property
+    def view(self):
+        """
+        getter returning the XML model view
+
+        returns : XML block
+        -------
+            XML model view to be parsed
+            by different methods
+        """
+        return self._xml_view
+
+    def get_instance_by_role(self, dmrole, all_instances=False):
+        """
+        Return the instance matching with @dmrole.
+        If all_instances is False, return the first INSTANCE matching with @dmrole.
```

Comment:
Duplicate text present in return description too.

Reply:
Duplicated text removed